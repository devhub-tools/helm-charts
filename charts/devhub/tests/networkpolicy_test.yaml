suite: NetworkPolicy
templates:
  - networkpolicy.yaml
release:
  name: devhub
  namespace: devhub

tests:
  - it: renders network policy with default values
    set:
      networkPolicy:
        create: true
    asserts:
      - matchSnapshot: {}

  - it: renders network policy with custom values
    set:
      networkPolicy:
        create: true
        ingress:
          - ports:
              - port: 4000
        egress:
          - ports:
              - port: 4000
    asserts:
      - matchSnapshot: {}

  - it: should not create network policy by default
    asserts:
      - hasDocuments:
          count: 0