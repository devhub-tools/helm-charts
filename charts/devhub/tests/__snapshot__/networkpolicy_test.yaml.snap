renders network policy with custom values:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/instance: devhub
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: devhub
        helm.sh/chart: devhub-2.1.0
      name: devhub
    spec:
      egress:
        - ports:
            - port: 4000
      ingress:
        - ports:
            - port: 4000
      podSelector: {}
      policyTypes:
        - Ingress
        - Egress
renders network policy with default values:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app.kubernetes.io/instance: devhub
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: devhub
        helm.sh/chart: devhub-2.1.0
      name: devhub
    spec:
      egress:
        - ports:
            - port: 53
              protocol: TCP
            - port: 53
              protocol: UDP
          to:
            - namespaceSelector:
                matchLabels:
                  kubernetes.io/metadata.name: kube-system
        - ports:
            - protocol: TCP
          to:
            - ipBlock:
                cidr: 0.0.0.0/0
      ingress:
        - ports:
            - port: 4000
              protocol: TCP
      podSelector: {}
      policyTypes:
        - Ingress
        - Egress
